lib/Dialect/MSFT/MSFTGenerator.cpp

```diff
@@ -0,0 +1,121 @@
+//===- MSFTGenerator.cpp - Implement MSFT generators ----------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Dialect/MSFT/MSFTDialect.h"
+
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Pass/PassRegistry.h"
+
+#include "llvm/ADT/SmallString.h"
+#include "llvm/ADT/StringMap.h"
+
+using namespace circt;
+using namespace msft;
+
+namespace {
+/// Holds the set of registered generators for each operation.
+class OpGenerator {
+  llvm::StringMap<GeneratorCallback> generators;
```

Comment:
Why not just a vector of callbacks? Is the intent for future logic to use the generatorName to select between multiple callbacks?

Reply:
I see the docs hint at that.